packages/core-app-api/api-report.md

```diff
@@ -264,6 +265,7 @@ export type AuthApiCreateOptions = {
   discoveryApi: DiscoveryApi;
   environment?: string;
   provider?: AuthProviderInfo;
+  configApi: ConfigApi;
```

Comment:
I had another think about this and in the end I think it's unfortunate to make this breaking change. Let's make this optional instead, and if it's not present we assume that the flag isn't set.

So to be clear, this would be `configApi?: ConfigApi` instead, along with all other similar usages. I think it's desirable enough to have it present that we should call it a deprecation though, so let's log a deprecation warning if the integrator does not supply this API, similar to what we do [here](https://github.com/backstage/backstage/blob/282d70fb1a40b20e025aa26ab88687642f312cca/packages/core-app-api/src/routing/FlatRoutes.tsx#L75-L78)